### PR TITLE
optionally disable deprecated apis

### PR DESCRIPTION
**Pull request by pauldreik**
This is a shot at fixing #1264.

I think that the deprecated apis should be off by default, but for now I set them to be on by default.

The fuzzers disable the deprecated apis.

I have tested that the amalgamation still works with/without the deprecated api.

### CODE REVIEW

File: include/simdjson/dom/parsedjson_iterator.h [hunk 1]
```diff
@@ -266,5 +267,6 @@ class [[deprecated("Use the new DOM navigation API instead (see doc/basics.md)")
 };
 
 } // namespace simdjson
+#endif //#if SIMDJSON_ENABLE_DEPRECATED_API
```
Comment (jkeiser) [inline include/simdjson/dom/parsedjson_iterator.h:270]: Remove the #if, I think--generally we don't include that in the #endif matching comment.

**Comment by jkeiser** [comment]
Because users use the amalgamated headers raw, without cmake, we need it to default to on, with something like

```c++
#ifndef SIMDJSON_ENABLE_DEPRECATED_API
#define SIMDJSON_ENABLE_DEPRECATED_API 1
#endif
```

That, or call it SIMDJSON_DISABLE_DEPRECATED_API :)

File: tests/basictests.cpp [hunk 2]
```diff
@@ -522,12 +525,16 @@ namespace parse_api_tests {
     return parser_moving_parser() &&
            parser_parse() &&
            parser_parse_many() &&
+#ifdef SIMDJSON_ENABLE_DEPRECATED_API
```
Comment (jkeiser) [inline tests/basictests.cpp:528]: Goodness, we *deeply* need to use a real test framework :)

**Review by jkeiser — CHANGES_REQUESTED** [review]
With the minor change of making sure it defaults to on without cmake, I'm super happy with this! It makes a ton of sense. Thanks :)

File: include/simdjson/dom/parsedjson_iterator.h (hunk 1 above)
Comment (pauldreik) [reply include/simdjson/dom/parsedjson_iterator.h:270]: sure thing!

File: tests/basictests.cpp (hunk 2 above)
Comment (pauldreik) [reply tests/basictests.cpp:528]: Yes....

**Comment by pauldreik** [comment]
> With the minor change of making sure it defaults to on without cmake, I'm super happy with this! It makes a ton of sense. Thanks :)

Thanks for the review, I fix these things and merge.

File: tests/basictests.cpp (hunk 2 above)
Comment (friendlyanon) [reply tests/basictests.cpp:528]: Just a suggestion, but [doctest](https://github.com/onqtam/doctest) is pretty good!

**Pull request merged**